tasty-query/shared/src/main/scala/tastyquery/Classpaths.scala

```diff
@@ -0,0 +1,50 @@
+package tastyquery
+
+object Classpaths:
+  /** Contains class data and tasty data. */
+  final class PackageData(val dotSeparatedName: String, val classes: IArray[ClassData], val tastys: IArray[TastyData]):
+    override def toString(): String = s"PackageData($dotSeparatedName)"
+
+  /** Contains class bytes.
+    *
+    * `binaryName` is the file name without the `.class` extension.
+    */
+  final class ClassData(val binaryName: String, val debugPath: String, val bytes: IArray[Byte]):
+    override def toString(): String = s"ClassData($binaryName, $debugPath)"
+
+  /** Contains tasty bytes.
+    *
+    * `binaryName` is the file name without the `.class` extension.
+    */
+  final class TastyData(val binaryName: String, val debugPath: String, val bytes: IArray[Byte]):
+    override def toString(): String = s"TastyData($binaryName, $debugPath)"
+
+  final class Classpath private (val packages: IArray[PackageData]):
+    def withFilter(binaryNames: List[String]): Classpath =
```

Comment:
`Paths.rootClassName` should re-encode name components before calling this method

Reply:
Done.